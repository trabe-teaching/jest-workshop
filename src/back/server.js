import path from "path";
import express from "express";
import cors from "cors";
import morgan from "morgan";
import createDataStore from "./store/datastore";
import * as logic from "./logic";

export default () => {
  const store = createDataStore();

  return express()
    .use(morgan("tiny"))
    .use(cors({ origin: "http://localhost:8080" }))
    .use("/", express.static(path.join(__dirname, "..", "front")))
    .get("/force-users", (req, res) => {
      const users = logic.findUsers(store);
      res.send(users);
    })
    .put("/seek/:id", (req, res) => {
      const users = logic.seekMaster(store, Number(req.params.id));
      res.send(users);
    })
    .put("/turn/:id", (req, res) => {
      const users = logic.turn(store, Number(req.params.id));
      res.send(users);
    })
    .put("/betray/:id", (req, res) => {
      const users = logic.betray(store, Number(req.params.id));
      res.send(users);
    })
    .put("/banish/:id", (req, res) => {
      const users = logic.banish(store, Number(req.params.id));
      res.send(users);
    })
    .post("/reset", (req, res) => {
      store.reset();
      res.send({});
    });
};
