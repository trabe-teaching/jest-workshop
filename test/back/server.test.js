import request from "supertest";
import server from "back/server";

jest.mock("back/logic");

describe("server responses", () => {
  describe("GET /force-users", () => {
    it("returns the users", done => {
      request(server())
        .get("/force-users")
        .expect(200)
        .expect("Content-Type", /json/)
        .expect("CORS", "http://localhost:8080")
        .expect("MORGAN", "tiny")
        .then(res => expect(JSON.parse(res.text)).toMatchSnapshot())
        .then(() => done());
    });
  });
});
