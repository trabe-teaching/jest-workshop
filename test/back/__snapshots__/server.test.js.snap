// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server responses GET /force-users returns the users 1`] = `
Array [
  "user1",
  "user2",
]
`;
